File: 1537-maximum-score-after-splitting-a-string/1537-maximum-score-after-splitting-a-string.java
class Solution {
    public int maxScore(String s) {
        int oneCount = (int) s.chars().filter(c -> c=='1').count();
		int zeroCount = 0;
		int ans  = 0;

		for(int i=0; i< s.length()-1; i++){
			char c = s.charAt(i);
			if(c == '0')
				zeroCount++;
			else
				oneCount--;
            ans = Integer.max(zeroCount + oneCount, ans);
		}
		return ans;
    }
}

File: 1537-maximum-score-after-splitting-a-string/1537-maximum-score-after-splitting-a-string.cpp
class Solution {
public:
    int maxScore(string s) {
        int one = 0, n = s.size()-1;
        int zero = 0, ans =0;
        for(auto ch : s)
            if(ch == '1') one++;
        for(int i=0; i<n; i++){
            if(s[i] == '1')
                one--;
            else
                zero++;
            ans = max(ans, one + zero);
        }
        return ans;
    }
};